### ISSUE TITLE
Closures fail to parse

### ISSUE THREAD

**Issue by segiddins** [trill-lang/trill#48]
The following snippet should work, but right now all closures require a function signature and that signature is assumed to be inside the closure?

```swift
func main() {
  let f: () -> Void = {
    return
  }
  println(f)
}
```

**Comment by harlanhaskins** [comment]
Right. That's a parse error, bough. I forgot that we require named parameters 😅
